### ISSUE TITLE
Enable HEIF image uploads

### ISSUE THREAD

**Issue by goobertron** [diaspora/diaspora#8355]
Please add HEIF to the list of image formats that can be uploaded to posts. Thank you!

**Comment by SuperTux88** [comment]
it would need to be processed on the server anyway to get the different sizes, but it's also not supported by all servers (I don't know if that's because it's a proprietary format, or if there are other reasons for that) ... so at least at the moment it's not that easy to support.

**Comment by goobertron** [comment]
Thanks, both. I'd hoped this would be a really simple addition. I'll leave the ticket open in case anyone feels like taking this on in the future – as long as you don't think it would hamper server performance.

**Comment by denschub** [comment]
Just for the record,

> I don't know if that's because it's a proprietary format, or if there are other reasons for that

ImageMagick supports HEIF if `libheif` is installed. A package is available [on Debian since Buster](https://packages.debian.org/buster/libheif1), and [on Ubuntu since Bionic](https://packages.ubuntu.com/bionic/libheif1), so it would not be unreasonable to add it as a dependency, imho.

(it's also not proprietary. :))

**Comment by SuperTux88** [comment]
At least for Ubuntu bionic I only find guides which include compiling imagemagick yourself, so I thought it's not just enough to install the lib? But maybe these guides are wrong? Later version seem to support it out of the box.

**Comment by denschub** [comment]
Uh, okay, I actually don't know if there are any additional dependencies, like a minimal required ImageMagick version. The IM website just says "it's supported", but not since when. If it's not supported in at least Focal, than yeah, this might simply not be realistic at the moment.

**Comment by SuperTux88** [comment]
I guess the lib needs to be installed at build time so it can link against it, that's why newer versions have it as hard dependency and then it's supported, older Ubuntu versions don't have it as dependency and that are also the ones with guides that describe recompiling imagemagick after installing the lib.

**Comment by denschub** [comment]
Yeah that makes total sense. I'm not entirely sure why I didn't check the build configs for the packages in question. Looking at the Changelog of the Debian package, I found [this change from 2018](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=914120):

> imagemagick (8:6.9.10.23+dfsg-1) unstable; urgency=high
>
>   [...]
>   * Enable HEIF/HEIC image format support (Closes: #914120).
>   * Enable WEBP image format (Closes: #806425, #912777)

ImageMagick 6.9.10.23 or newer is in Ubuntu since Focal, and in Debian since Buster.

Debian Stretch is pretty close to EOL, but unfortunately Bionic is still supported. Requiring Ubuntu Focal or newer *might* be reasonable, but I'm not going to make that call. I'd tend towards being fine with that, especially since the same release also enables WebP, and diaspora\* could gain a lot by converting to WebP instead of JPEG...
